File: Medii de Proiectare si Programare/TrackAndField-C#/TrackAndField/repository/TrackRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using log4net;
using TrackAndField.model;

namespace TrackAndField.repository
{
    class TrackRepository : ITrackRepository
    {
        private static readonly ILog logger = LogManager.GetLogger("TrackRepository");

        public TrackRepository()
        {
            logger.Info("Initializing TrackRepository class");
        }

        public IEnumerable<Track> FindAll()
        {
            logger.InfoFormat("Finding all tracks...");

            IList<Track> tracks = new List<Track>();

            var connection = DBUtils.GetConnection();

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "SELECT * FROM Track";

                using (var dataReader = command.ExecuteReader())
                {
                    while (dataReader.Read())
                    {
                        string name = dataReader.GetString(0);
                        int minimumAge = dataReader.GetInt32(1);
                        int maximumAge = dataReader.GetInt32(2);
                        Track track = new Track(name, minimumAge, maximumAge);
                        tracks.Add(track);
                    }
                }
            }

            return tracks;
        }

        public Track FindOne(string id)
        {
            Track track = null;

            var connection = DBUtils.GetConnection();

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "SELECT * FROM Track WHERE name = @id";

                var idParameter = command.CreateParameter();
                idParameter.ParameterName = "@id";
                idParameter.Value = id;
                command.Parameters.Add(idParameter);

                using (var dataReader = command.ExecuteReader())
                {
                    if (dataReader.Read())
                    {
                        string name = dataReader.GetString(0);
                        int minimumAge = dataReader.GetInt32(1);
                        int maximumAge = dataReader.GetInt32(2);
                        track = new Track(name, minimumAge, maximumAge);
                        logger.InfoFormat("Track found: {0}", track);
                    }
                }
            }

            return track;
        }

        public Track Save(Track entity)
        {
            var connection = DBUtils.GetConnection();

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "INSERT INTO Track(name, minimumAge, maximumAge) VALUES (@name, @min, @max)";

                var nameParameter = command.CreateParameter();
                nameParameter.ParameterName = "@name";
                nameParameter.Value = entity.Name;
                command.Parameters.Add(nameParameter);

                var minParameter = command.CreateParameter();
                minParameter.ParameterName = "@min";
                minParameter.Value = entity.MinimumAge;
                command.Parameters.Add(minParameter);

                var maxParameter = command.CreateParameter();
                maxParameter.ParameterName = "@max";
                maxParameter.Value = entity.MaximumAge;
                command.Parameters.Add(maxParameter);

                var result = command.ExecuteNonQuery();
            }
            return entity;
        }
    }
}

File: Medii de Proiectare si Programare/TrackAndField-Java/src/main/java/repository/TrackRepository.java
package repository;

import model.Track;
import model.User;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import utils.JdbcUtils;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class TrackRepository implements ITrackRepository {
    private static final Logger logger= LogManager.getLogger();
    private JdbcUtils dbUtils;

    public TrackRepository(Properties props) {
        logger.info("Initializing TrackRepository with properties: {} ",props);
        dbUtils = new JdbcUtils(props);
    }

    @Override
    public Track findOne(String s) {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        Track track = null;
        try(PreparedStatement preStmt = con.prepareStatement("select * from Track where name = (?)")){
            preStmt.setString(1, s);
            try(ResultSet result = preStmt.executeQuery()){
                int minimumAge = result.getInt("minimumAge");
                int maximumAge = result.getInt("maximumAge");
                track = new Track(s, minimumAge, maximumAge);
            }
        }catch (SQLException ex)
        {
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
        return track;
    }

    @Override
    public Iterable<Track> findAll() {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        List<Track> tracks = new ArrayList<>();
        try(PreparedStatement preStmt = con.prepareStatement("select * from Track")){
            try(ResultSet result = preStmt.executeQuery()){
                while(result.next()){
                    String name = result.getString("name");
                    int minimumAge = result.getInt("minimumAge");
                    int maximumAge = result.getInt("maximumAge");
                    Track track = new Track(name, minimumAge, maximumAge);
                    tracks.add(track);
                }
            }
        }catch (SQLException ex)
        {
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
        return tracks;
    }

    @Override
    public Track save(Track entity) {
        logger.traceEntry("Saving track {}", entity);
        Connection connection = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = connection.prepareStatement("insert into Track(name, minimumAge, maximumAge) values (?, ?, ?)")){
            preparedStatement.setString(1, entity.getName());
            preparedStatement.setInt(2, entity.getMinimumAge());
            preparedStatement.setInt(3, entity.getMaximumAge());
            int result = preparedStatement.executeUpdate();
            logger.trace("Saved {} instances", result);
        }catch(SQLException ex)
        {
            logger.error(ex);
            System.err.print("Error DB" + ex);
        }
        logger.traceExit();
        return entity;
    }
}
